### PR TITLE
Set endpoint during the upgrade

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1225

### CODE REVIEW

File: longhorndev/dev.go [hunk 1]
```diff
@@ -239,6 +239,9 @@ func (d *LonghornDevice) PrepareUpgrade() error {
 }
 
 func (d *LonghornDevice) FinishUpgrade() (err error) {
+	d.Lock()
```
Comment (yasker) [inline longhorndev/dev.go:242]: The newly added code is basically duplicated of `Start()` without creating target etc. Can we call a common function from Start() instead?

File: longhorndev/dev.go [hunk 2]
```diff
@@ -239,45 +258,16 @@ func (d *LonghornDevice) PrepareUpgrade() error {
...
-	bsOpts := fmt.Sprintf("size=%v", d.size)
-	scsiDev, err := iscsidev.NewDevice(d.name, d.GetSocketPath(), "longhorn", bsOpts)
-	if err != nil {
-		return err
-	}
-	d.scsiDevice = scsiDev
-
-	return nil
+	// TODO: Need to fix `ReloadSocketConnection` since it doesn't work for frontend `FrontendTGTISCSI`.
+	return d.ReloadSocketConnection()
```
Comment (yasker) [inline longhorndev/dev.go:266]: Should it be reload first then start device? If reload failed, then we cannot start the device.

File: longhorndev/dev.go [hunk 3]
```diff
@@ -239,45 +258,16 @@ func (d *LonghornDevice) PrepareUpgrade() error {
...
 }
 
 func (d *LonghornDevice) ReloadSocketConnection() error {
-	d.RLock()
```
Comment (yasker) [inline longhorndev/dev.go:278]: Why remove the lock here?

File: longhorndev/dev.go (hunk 2 above)
Comment (shuo-wu) [reply longhorndev/dev.go:266]: But we need to wait for the socket ready first before reloading... That's why I choose to start the device first.

File: longhorndev/dev.go (hunk 3 above)
Comment (shuo-wu) [reply longhorndev/dev.go:278]: Oh... I Forgot to add the lock back.

File: longhorndev/dev.go (hunk 2 above)
Comment (yasker) [reply longhorndev/dev.go:266]: Oh, probably need the d.getDev() for the reload socket call?

Comment (yasker) [reply longhorndev/dev.go:266]: Don't need to merge the two parts of logic into start() though. You can leave wait for socket here still.

Comment (yasker) [reply longhorndev/dev.go:276]: Now we can reload first then init the device later.

**Pull request merged**